src/main/java/gregtech/api/recipes/logic/ParallelLogic.java

```diff
@@ -2,153 +2,229 @@
+        int itemMultiplier = getMaxRatioItem(ingredientStacks, recipe, parallelAmount);
+        // Find the maximum number of recipes that can be performed from the fluids in the fluid input inventories
+        int fluidMultiplier = getMaxRatioFluid(fluidStacks, recipe, parallelAmount);
 
         // Find the maximum number of recipes that can be performed from all available inputs
-        int minMultiplier = Math.min(itemMultiplier, fluidMultiplier);
+        return Math.min(itemMultiplier, fluidMultiplier);
+    }
 
-        // No fluids or items were found in the input inventories that match the recipe's inputs
-        if(minMultiplier == Integer.MAX_VALUE) {
-            return null;
+    /**
+     *
+     * @param recipe The recipe
+     * @param outputs the item output inventory
+     * @param fluidOutputs the fluid output tanks
+     * @param parallelAmount the maximum expected amount
+     * @return returns the amount of recipes that can be merged successfully into a given output inventory
+     */
+    public static int limitByOutputMerging(Recipe recipe, IItemHandlerModifiable outputs, IMultipleTankHandler fluidOutputs, int parallelAmount) {
+        int maxMultiplierItems = parallelAmount;
+        int maxMultiplierFluids = parallelAmount;
+        if (recipe.getOutputs().size() > 0) {
+            maxMultiplierItems = limitParallelByItems(recipe, new OverlayedItemHandler(outputs), parallelAmount);
+            if (maxMultiplierItems == 0) {
+                return 0;
+            }
         }
-
-        // Create holders for the various parts of the new multiplied Recipe
-        List<CountableIngredient> newRecipeInputs = new ArrayList<>();
-        List<FluidStack> newFluidInputs = new ArrayList<>();
-        List<ItemStack> outputItems = new ArrayList<>();
-        List<FluidStack> outputFluids = new ArrayList<>();
-
-        // Populate the various holders of the multiplied Recipe
-        this.multiplyInputsAndOutputs(newRecipeInputs, newFluidInputs, outputItems, outputFluids, recipe, minMultiplier);
-
-        // Build the new Recipe with multiplied components
-        RecipeBuilder<?> newRecipe = recipeMap.recipeBuilder()
-                .inputsIngredients(newRecipeInputs)
-                .fluidInputs(newFluidInputs)
-                .outputs(outputItems)
-                .fluidOutputs(outputFluids)
-                .EUt(recipe.getEUt())
-                .duration(recipe.getDuration());
-
-        // Add the chanced outputs to the multiplied recipe
-        copyChancedItemOutputs(newRecipe, recipe, minMultiplier);
-
-        // Return the multiplied Recipe
-        return newRecipe.build().getResult();
+        if (recipe.getFluidOutputs().size() > 0) {
+            maxMultiplierFluids = limitParallelByFluids(recipe, new OverlayedFluidHandler(fluidOutputs), parallelAmount);
+        }
+        return Math.min(maxMultiplierItems, maxMultiplierFluids);
```

Comment:
If `maxMultiplierItems` is used above, `maxMultiplierFluids` could be returned without the `Math.min()` call since the fluids multiplier should be guaranteed to be less than or equal to the item parallel, if I understand the logic correctly.